volttron/platform/agent/base_historian.py

```diff
@@ -1572,30 +1587,43 @@ def get_outstanding_to_publish(self, size_limit):
             value = loads(row[4])
             headers = {} if row[5] is None else loads(row[5])
             meta = self._meta_data[(source, topic_id)].copy()
+            topic = self._backup_cache[topic_id]
+
+            # check for duplicates before appending row to results
+            if (topic, timestamp) in unique_records:
```

Comment:
why not  use just topic_id, timestamp ?

Reply:
Good question. I originally used topic because I was using the solution from [@acedrew's earlier PR](https://github.com/VOLTTRON/volttron/pull/2443/files#diff-4d58a5e8d1e986cea80064a795a19e46d509c2a22d449c116651a121d3df514cR187), which doesn't use topic_id because the concrete historian doesn't have topic_id, only topic, which makes sense because topic_id comes from the cache, which only the base historian knows, but not a concrete historian. 

I've changed the code accordingly and all tests still pass. Topic_id is more concise to identify duplicates.